### PR TITLE
Bug 1388337 - Clear lifecycle hooks when changing strategy from Recreate to Rolling

### PR DESCRIPTION
**Pull request by jhadvig**
@spadgett talked to @mfojtik && @kargakis regarding this and we agreed that when  switching between rolling<->recreate strategy it doesnt make sense to preserve the old params.
PTAL

### CODE REVIEW

**Comment by spadgett** [comment]
@jhadvig I'm unclear on what parameters should be moved when. Can you summarize the different cases?

**Comment by jhadvig** [comment]
@spadgett so user can move parameters when switching between rolling and recreate strategy, since those two strategies have 3 common parameters (timeout, pre lifecycle hook, post lifecycle hook). If user switches from Rolling to Recreate (or vice versa) he will be prompt with dialog if the common params from the original strategy shall be moved to the selected one. If so we will copy them. But for both scenarios(1. moving the params to the selected strategy, 2. not moving params) we shall remove the parameters from the original strategy, since they are not necessary for the DC.
On the other hand if switching from Rolling || Recreate to Custom(or vice versa) it makes sense to keep the original strategy parameters.

**Comment by spadgett** [comment]
OK, so the behavior moving the parameters is the same as before. The change is to delete the parameters from the previous strategy when making a rolling <-> recreate switch.

**Comment by jhadvig** [comment]
@spadgett yes

**Comment by spadgett** [comment]
[merge]
